control-agent/text-to-speech/speech-detection.py

```diff
@@ -132,26 +133,82 @@ def QnA(question):
 
 run = True
 
+speech.speak("starting up")
+
 while run == True:
 
     intent = recognize_intent()
     try:
         if intent.intent_id == "Vision":
             speech.speak("I'm looking")
-            result = speech.recognize(
-                "https://robots.ieee.org/robots/stretch/stretch-1200x630.jpg"
-            )
 
-            speech.speak(
-                "I can see" + str(result["description"]["captions"][0]["text"])
-            )
+            image = realsense.take_photo()
+
+            result = speech.recognize(image)
+
+            if (
+                "person"
+                or "boy"
+                or "man"
+                or "woman" in result["description"]["captions"][0]["text"]
+            ):
+                # TODO more than one person?
+                try:
+                    person = speech.identify_face(
+                        speech.recognize_face(image)[0]["faceId"]
```

Comment:
If you break this out into a function, you could iterate over the  results of speech.recognize_face. E.g. for face in recognition_results: append_name

https://stackoverflow.com/questions/53981845/grammatically-correct-human-readable-string-from-list-with-oxford-comma

Reply:
I'll raise another issue